Formula/libgweather.rb

```diff
@@ -1,10 +1,17 @@
 class Libgweather < Formula
   desc "GNOME library for weather, locations and timezones"
   homepage "https://wiki.gnome.org/Projects/LibGWeather"
-  url "https://download.gnome.org/sources/libgweather/40/libgweather-40.0.tar.xz"
-  sha256 "ca4e8f2a4baaa9fc6d75d8856adb57056ef1cd6e55c775ba878ae141b6276ee6"
+  url "https://download.gnome.org/sources/libgweather/4.2/libgweather-4.2.0.tar.xz"
+  sha256 "af8a812da0d8976a000e1d62572c256086a817323fbf35b066dbfdd8d2ca6203"
   license all_of: ["GPL-2.0-or-later", "LGPL-2.1-or-later"]
-  revision 1
+  version_scheme 1
+
+  # Ignore version 40 which is older than 4.0 as discussed in
+  # https://gitlab.gnome.org/GNOME/libgweather/-/merge_requests/120#note_1286867
+  livecheck do
+    url :stable
+    regex(/libgweather[._-]v?((?:[0-35-9]\d*|4|4[1-9]\d*|40\d+)\.([0-8]\d*?)?[02468](?:\.\d+)*?)\.t/i)
```

Comment:
```suggestion
    strategy :gnome do |page, regex|
      page.scan(regex).select { |match| Version.new(match.first) < 40 }.flatten
    end
```

> I added a regex to reject 40.* versions. Could be more readable to use a block to reject specific 40.0 instead.

I'm in favor of using a `strategy` block to omit versions >= 40 in this case, as using a regex will prevent the `Gnome` strategy from using its internal logic for enforcing the GNOME version scheme (requiring an overly complicated regex like this). As you suggested, the `strategy` block approach is considerably easier to comprehend (and, consequently, maintain).